### PR TITLE
#62 Make gamepad API checks static. This is useful to first check if the …

### PR DESCRIPTION
**Pull request by danrossi**
…browser supports the api.

### CODE REVIEW

File: demo.js [hunk 1]
```diff
@@ -48,6 +48,12 @@ var gamepadConfig = {
   }
 };
 
+
+
```
Comment (cvan) [inline demo.js]: can you remove these extra newlines?

File: demo.js [hunk 2]
```diff
@@ -48,6 +48,12 @@ var gamepadConfig = {
...
+console.log("Has gamepads API support ", Gamepads.hasGamepads());
```
Comment (cvan) [inline demo.js]: can you change this to

``` js
console.log('Gamepads detected', Gamepads.hasGamepads());
```

File: src/lib/gamepads.js [hunk 3]
```diff
@@ -88,6 +88,15 @@ export default class Gamepads extends EventEmitter {
     }
   }
 
+  /**
+   * Make gamepads API support check static
```
Comment (cvan) [inline src/lib/gamepads.js]: can you add a period?

File: src/lib/gamepads.js [hunk 4]
```diff
@@ -88,6 +88,15 @@ export default class Gamepads extends EventEmitter {
...
+   * @returns {string[]}
+   */
+  static get gamepadApis() {
+    return ['getGamepads', 'webkitGetGamepads', 'webkitGamepads'];
+  }
+
+
```
Comment (cvan) [inline src/lib/gamepads.js]: can you remove newline?

File: src/lib/gamepads.js [hunk 5]
```diff
@@ -134,24 +143,40 @@ export default class Gamepads extends EventEmitter {
...
     }
     return false;
   }
 
-  _getGamepads() {
-    for (var i = 0, len = this._gamepadApis.length; i < len; i++) {
-      if (this._gamepadApis[i] in navigator) {
-        return navigator[this._gamepadApis[i]]();
+  /**
+   * Make gamepads API support static
```
Comment (cvan) [inline src/lib/gamepads.js]: period please

File: src/lib/gamepads.js [hunk 6]
```diff
@@ -134,24 +143,40 @@ export default class Gamepads extends EventEmitter {
...
+   * @returns {*}
```
Comment (cvan) [inline src/lib/gamepads.js]: can you define the type and provide a description?

**Comment by danrossi** [comment]
I've cleaned up the code and comments. However what type is documented to be returned ?. 

I put 

```
@returns {Gamepad}
```

**Comment by danrossi** [comment]
Apologies I had to consult the jsdocs it's 

```
@returns {Gamepad[]}
```

**Comment by cvan** [comment]
I can rebuild after merged. thanks!

**Pull request merged**